### DISCUSSION TITLE
TypeScript compiler support

### DISCUSSION THREAD

**Discussion by patriksvensson** [cake-build/cake#3509] [Extension requests]

**Comment by RichiCoder1** [comment]
This will either TCS to be assumed as installed after review. Assuming that's installed, following a https://github.com/k-maru/grunt-typescript scheme should work.

``` C#
TypeScript(new TypeScriptSettings {
        Source = "/src/UI/scripts/**/*.ts",
        Destination = "/build/scripts/*.js".
        // Should we do Options = new TypeScript options, or just have them as part of Settings?
    });
```

``` C#
TypeScript(settings => {
        settings.Source("src/UI/scripts/**/*.ts");
        settings.Destination("build/scripts/*.js");
        settings.Module("AMD");
        settings.SourceMap(true);
    });
```

**Comment by RichiCoder1** [comment]
I may chew on this for unrelated reasons. Adding an addendum that with the upcoming TS 1.5, there's now a tsconfig.json which can contain many of these settings.

**Comment by devlead** [comment]
Cool @RichiCoder1 I would suggest you start as an addin to try out the concept.

**Comment by RichiCoder1** [comment]
Absolutely. I've got something prototyped out.

**Comment by RichiCoder1** [comment]
Probably will grab this with my Node and NPM items.

**Comment by klaus-liebler** [comment]
I am stronly interested in this feature. Does anybody plan to implement this (soon)?
